calico/felix/profilerules.py

```diff
@@ -282,19 +290,16 @@ def _update_chains(self):
         self._iptables_updater.rewrite_chains(updates, deps, async=False)
 
 
-def extract_tags_from_profile(profile):
+def extract_tags_and_selectors_from_profile(profile):
     if profile is None:
         return set()
-    tags = set()
+    tags_and_sels = set()
     for in_or_out in ["inbound_rules", "outbound_rules"]:
         for rule in profile.get(in_or_out, []):
-            tags.update(extract_tags_from_rule(rule))
-    return tags
-
-
-def extract_tags_from_rule(rule):
-    return set(rule[key] for key in ["src_tag", "dst_tag"]
-               if key in rule and rule[key] is not None)
+            for key in ["src_tag", "dst_tag", "src_selector", "dst_selector"]:
+                if key in rule:
+                    tags_and_sels.add(rule[key])
```

Comment:
Lost a

    rule[key] is not None

condition here - is that intentional?  Also I don't understand why you changed from the set comprehension - that seemed nicer to me.